### ISSUE TITLE
some .SH script   likely to recursively delete ALL java code from other projects ?

### ISSUE THREAD

**Issue by anonym2000** [deeplearning4j/deeplearning4j#6356]
-running system:  macos sierra 10.0.12
- code:  deeplearning 0.9.1

**possible** issue: 
1)   in a dir called "projects" I have several  intellij projects .
2) in the same dir I cloned the "deeplearning"    project from github 
3) after running ./perform-release.sh  or ./createmono.sh  inside the deeplearning dir  ( not sure which one did the mess, I just noticed later )  ALL  java dirs inside EACH intellij  project was gone, even though they had nothing to do with DL4j.

I saw that at "createmono.sh"  some recursive delete ( -rf  )  stuff is called.  There is also a variable something like $PROJECT_DIR" -  I think there is a variable like that in intellij too ..

if you confirm this issue please inform me at gitter :  @javadev-berlin   ( please do not use github or email )

**Comment by AlexDBlack** [comment]
Uh, not sure what to tell you here.

You can look at both of those scripts here:
https://github.com/deeplearning4j/deeplearning4j/blob/master/perform-release.sh
https://github.com/deeplearning4j/deeplearning4j/blob/master/createmono.sh

The 'createmono.sh' script is the only one of the two that removes directories, but it's only those with the specified names: https://github.com/deeplearning4j/deeplearning4j/blob/master/createmono.sh#L25
So it would seem unlikely to be the cause here.

Of course, it begs the question why you would be running either of these scripts at all. Or even cloning the repo in the first place (>99% of users don't need to). These scripts are for internal development and release use, and aren't mentioned in our docs at all.

**Comment by anonym2000** [comment]
_" ...Of course, it begs the question why you would be running either of these scripts at all."_ 
Those scripts are **public** . The code repo is **public**.   Why I ran those is not the point.   The point is that THERE MIGHT BE a potentially dangerous script in the code your company put out there.    What I ( and every developer who wants to work with your code ) would expect is that Skymind ( or whoever is behind that repo ) does not release any script that could delete ALL your java projects  ( it's like a hidden trap: click me and I will destroy all your code ) ... just imagine having all your work of months deleted forever in seconds, gone forever.    

  Wouldn't  even the remove possibility of this being the case worth a closer look / tests to see if that can happen ?

**Comment by AlexDBlack** [comment]
I'm sorry that you lost your files, but I'm quite skeptical that any DL4J repo code is at fault here.
You mentioned two scripts.

perform-release is a combination of sed and mvn. We use this script for performing releases - neither of which will delete arbitrary files. We also use this for releases.

As for createmono.sh: I tested this on my local Ubuntu 17.10 machine, with deeplearning4j repo parallel to 5 other java/maven directories (i.e., same parent directory as deeplearning4j repo). As expected, it performed changes on my deeplearning4j repo, (operating on the directory names used in the for loop) but left all other projects untouched (and as far as I can tell, all other files on my system also). And that's assuming I manually force it to continue to completion, when it runs into merge conflicts and the like (as it does multiple times).
All that said, I'm in favor of deleting this script as it is no longer required.

I'm not aware of any other scripts in the repository that could have this effect either.
Of course, if this is the case, we definitely want to know.

To be safe, I'll do a more thorough review of the other scripts that appear in the repository (note I did carefully review both scripts that you mention before replying earlier, and I'm familiar with most of the scripts in the repository already). Beyond that, and without further information or any way to reproduce this behaviour (or even a hypothesis as to how it could happen), I have no choice but to consider this report as not actionable.
